File: re/GhidraRefsTo.java
// Print references to a single address in headless Ghidra.
//
// Usage:
//   analyzeHeadless ... -postScript GhidraRefsTo.java 0x361e

import ghidra.app.script.GhidraScript;
import ghidra.program.model.address.Address;
import ghidra.program.model.symbol.Reference;

public class GhidraRefsTo extends GhidraScript {
    private Address parseAddr(String text) throws Exception {
        String s = text.trim().toLowerCase();
        if (s.startsWith("0x")) {
            s = s.substring(2);
        }
        return currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(s);
    }

    @Override
    protected void run() throws Exception {
        String[] args = getScriptArgs();
        if (args.length < 1) {
            printerr("usage: GhidraRefsTo.java <addr>");
            return;
        }

        Address addr = parseAddr(args[0]);
        Reference[] refs = getReferencesTo(addr);
        println("REFS TO " + addr);
        for (Reference ref : refs) {
            println(ref.getFromAddress() + " " + ref.getReferenceType());
        }
    }
}

File: re/ghidra_refs_to.py
from ghidra.program.model.symbol import RefType


def parse_addr(text):
    s = text.strip().lower()
    if s.startswith("0x"):
        s = s[2:]
    return currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(s)


args = getScriptArgs()
if len(args) < 1:
    printerr("usage: ghidra_refs_to.py <addr>")
    exit(1)

target = parse_addr(args[0])
refs = list(getReferencesTo(target))

print("TARGET {}".format(target))
print("REFCOUNT {}".format(len(refs)))
for ref in refs:
    frm = ref.getFromAddress()
    kind = ref.getReferenceType()
    fn = getFunctionContaining(frm)
    fn_name = fn.getEntryPoint().toString() if fn else "<no-func>"
    print("{} {} {}".format(frm, fn_name, kind))
